core/application/userWorkflow/UserService.ts

```diff
@@ -78,54 +66,50 @@ export class UserService {
   async updateUser(
     userAttributes: UpdateUserAttributes,
     userRepository: Repository<UserDetailsDTO>,
-    locationRepository: Repository<LocationDTO>,
-    model: LocationModel
-  ): Promise<string> {
-    try {
-      const { userId, preferredDonationLocations, ...restAttributes } =
-        userAttributes
-      const updateData: Partial<UserDetailsDTO> = {
-        ...restAttributes,
-        id: userId,
-        updatedAt: new Date().toISOString()
-      }
+    locationRepository: LocationRepository<LocationDTO>
+  ): Promise<void> {
+    const { userId, preferredDonationLocations, ...restAttributes } = userAttributes
+    const updateData: Partial<UserDetailsDTO> = {
+      ...restAttributes,
+      id: userId,
+      updatedAt: new Date().toISOString()
+    }
 
-      updateData.age = this.calculateAge(userAttributes.dateOfBirth)
-      updateData.availableForDonation = this.calculateAvailableForDonation(
-        userAttributes.lastDonationDate,
-        userAttributes.availableForDonation
-      )
+    updateData.age = this.calculateAge(userAttributes.dateOfBirth)
+    updateData.availableForDonation = this.checkLastDonationDate(
+      userAttributes.lastDonationDate,
+      userAttributes.availableForDonation
+    )
 
-      await userRepository.update(updateData)
-      await this.updateUserLocation(
-        model,
-        userId,
-        locationRepository,
-        preferredDonationLocations,
-        updateData
-      )
-      return 'Updated your Profile info'
-    } catch (error) {
+    await userRepository.update(updateData).catch((error) => {
+      throw new UserOperationError(`Failed to update user. Error: ${error}`, GENERIC_CODES.ERROR)
+    })
+
+    await this.updateUserLocation(
+      userId,
+      preferredDonationLocations,
+      updateData,
+      locationRepository
+    ).catch((error) => {
       throw new UserOperationError(
-        `Failed to update user. Error: ${error}`,
+        `Failed to update user location. Error: ${error}`,
         GENERIC_CODES.ERROR
       )
-    }
+    })
   }
 
-  private calculateAvailableForDonation(
-    lastDonationDate: string,
+  private checkLastDonationDate(
+    lastDonationDate: string | undefined,
     availableForDonation: AvailableForDonation
   ): AvailableForDonation {
-    if (lastDonationDate !== '') {
+    if (lastDonationDate !== undefined && lastDonationDate !== '') {
       const donationDate = new Date(lastDonationDate)
       const currentDate = new Date()
 
       if (!isNaN(donationDate.getTime())) {
         const donationMonths = differenceInMonths(currentDate, donationDate)
-        return donationMonths >
-          Number(process.env.MIN_MONTHS_BETWEEN_DONATIONS)
-          ? 'yes'
+        return donationMonths > Number(process.env.MIN_MONTHS_BETWEEN_DONATIONS)
+          ? availableForDonation
           : 'no'
```

Comment:
Why we are setting string instead of boolean here?

Reply:
we are setting `AvailableForDonation` here which is `yes` or `no`

Reply:
same question from me as well

Reply:
created an issue: https://github.com/CraftsmenLtd/BloodConnect/issues/317